### PR TITLE
Trigger deploy of whatwg.org on changes

### PR DESCRIPTION
**Pull request by foolip**
Fixes https://github.com/whatwg/sg/issues/110.

### CODE REVIEW

File: .github/workflows/dispatch.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    branches:
+    - master
+jobs:
+  dispatch:
+    name: Dispatch
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: peter-evans/repository-dispatch@v1
+      with:
+        token: ${{ secrets.DISPATCH_TOKEN }}
```
Comment (foolip) [inline .github/workflows/dispatch.yml:14]: This token is a personal access token for @autofoolip, who I've given write access to this repository. That token has the public_repos scope.

Comment (foolip) [reply .github/workflows/dispatch.yml:14]: Actually, using https://github.com/whatbot would be better. I'll do that instead.

Comment (foolip) [reply .github/workflows/dispatch.yml:14]: Updated now.

**Comment by foolip** [comment]
This seems to work, https://github.com/whatwg/sg/runs/780908592 triggered https://github.com/whatwg/whatwg.org/runs/780910278 (as @whatbot)

File: .github/workflows/dispatch.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+on:
+  push:
+    branches:
+    - master
+jobs:
+  dispatch:
+    name: Dispatch
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: peter-evans/repository-dispatch@v1
```
Comment (annevk) [inline .github/workflows/dispatch.yml:12]: I guess Dependabot monitors these too?

Comment (foolip) [reply .github/workflows/dispatch.yml:12]: One can configure Dependabot to bump GitHub Actions, yes. I haven't tried it, though.

**Comment by foolip** [comment]
> So the risks here are that if peter-evans/repository-dispatch@v1 goes rogue the DISPATCH_TOKEN is compromised but that cannot really be used for much anyway?

An attacker would get write access to the whatwg/whatwg.org repo. From there they should be able to extract the deploy keys to marquee and SSH into it, at which point we'd be pretty thoroughly compromised.

That's a fair bit of trust to place in a nice-to-have wrapper. We could pin it to an exact commit, but reviewing changes to https://github.com/peter-evans/repository-dispatch/blob/v1/dist/index.js isn't going to be easy.

I'll try to craft the `curl` command line for this instead.

**Comment by foolip** [comment]
OK, after some flailing about I got it to work, https://github.com/whatwg/sg/runs/781040423 triggered https://github.com/whatwg/whatwg.org/actions/runs/138597128. @annevk r?

**Comment by foolip** [comment]
FWIW, Azure Pipelines has something lighter-weight where no agent at all is required:
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/phases?view=azure-devops&tabs=yaml#server-jobs

But GitHub Actions does not, AFAICT.

**Comment by foolip** [comment]
Alright, this all seems to work: https://github.com/whatwg/whatwg.org/runs/781299300

**Comment by foolip** [comment]
I did accidentally rebase all the junk commits onto master, though. Disabled branch protection and force pushed to fix.

**Pull request merged**